### PR TITLE
[deps] Bump gl-native to v10.10.0, common to v23.2.0.

### PR DESCRIPTION
**Pull request by pengdev**
## Features and improvements 🏁
* Update mapbox-common to v23.2.0.
* Eliminate tiles re-creation and re-layout on zooming map with globe, when the camera is trespassing the zoom projection border.
* Avoid tiles re-layout on enabling terrain with zero exaggeration.
* Added API to enable/disable rendering of world copies in mercator mode.
* Improve symbol filtering performance when `distance-from-camera` and `pitch` expressions are used.
* Reduce number of operations when terrain is used with zero exaggeration.

## Bug fixes
* Handle OOB when getting DEM Data
* Fix a rare precision issue with symbol and circle layer occlusion when terrain is enabled.
* Fixes a rare crash caused by a race condition during gesture handling.
* Disable location indicator occlusion testing with the terrain when exaggeration is set to 0. This fixes occasional location indicator disappearance.
* Fix an issue where queried symbol features did not contain associated feature state.
* Clear geojson tiles after the source is updated with empty features in order to eliminate "phantom tile" artefacts and to obviate extra work for keeping empty tiles.
* Eliminate duplicated tile async upload requests - fix a race condition causing a tile rendering the previous layout result.
* Truncate long single line text in symbol layers to prevent rendering artifact.
* Fixes an issue when the mixed usage of patterns and icons caused bleeding of textures.
* Fixes visible tile borders when MSAA enabled.
* Fix LOD and prevent from flickering on enabling terrain with zero exaggeration.
* Setting empty value for a style layer filter now clears the filter.
* Fixes rendering artifacts near tile borders when using terrain with zero exaggeration.
* Fixes fog rendering when terrain is used with zero exaggeration.
* Fix `distance-to-center` filtering of symbols when terrain is enabled.
* Improved `cameraForCoordinate` result quality with pitch and with terrain.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,12 +6,54 @@ Mapbox welcomes participation and contributions from everyone.
...
+* Fix an issue where location updates were not provided on correct thread/looper. ([1836](https://github.com/mapbox/mapbox-maps-android/pull/1836))
+* Dispatch location updates as a location event when the app is moving to background. ([1836](https://github.com/mapbox/mapbox-maps-android/pull/1836))
+* Fix `LocationEngine` to support multiple concurrent clients. ([1836](https://github.com/mapbox/mapbox-maps-android/pull/1836))
+* Fix immediate camera animation on API level 23 or below. ([1842](https://github.com/mapbox/mapbox-maps-android/pull/1842))
 * Fix loss of `Widget` during background/foreground transition. ([1864](https://github.com/mapbox/mapbox-maps-android/pull/1864))
 * Fix displaying MapView in Android Studio Layout Preview. ([1881](https://github.com/mapbox/mapbox-maps-android/pull/1881))
 * Fix a bug where `ViewAnnotationManager.annotations` API doesn't return correct camera bounds. ([1861](https://github.com/mapbox/mapbox-maps-android/pull/1861))
+* Handle OOB when getting DEM Data ([1891](https://github.com/mapbox/mapbox-maps-android/pull/1891))
```
Comment (kiryldz) [inline CHANGELOG.md:36]: was included in v10.9.1 patch, guess no need to specify it here

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,12 +6,54 @@ Mapbox welcomes participation and contributions from everyone.
 
 # 10.10.0
 ## Features ✨ and improvements 🏁
+* Introduce view annotation `ViewAnnotationManager.annotations` API to access list of added view annotations. ([1751](https://github.com/mapbox/mapbox-maps-android/pull/1751))
+* Introduce view annotation `ViewAnnotationManager.cameraForAnnotations` API to get camera options for given view annotations list. ([1753](https://github.com/mapbox/mapbox-maps-android/pull/1753))
+* Unify the `margin`/`translation` `Widget` APIs into the `WidgetPosition.offset`, rename `WidgetPosition.horizontal`/`WidgetPosition.vertical` to `WidgetPosition.horizontalAlignment`/`WidgetPosition.verticalAlignment`; Deprecate the original constructors and `setTranslation` APIs. ([1782](https://github.com/mapbox/mapbox-maps-android/pull/1782))
+* Add API for removing atmosphere. ([1841](https://github.com/mapbox/mapbox-maps-android/pull/1841))
 * Introduce `MapTelemetry.getUserTelemetryRequestState` API. ([1877](https://github.com/mapbox/mapbox-maps-android/pull/1877))
+* Update mapbox-common to v23.2.0. ([1891](https://github.com/mapbox/mapbox-maps-android/pull/1891))
```
Comment (kiryldz) [inline CHANGELOG.md:14]: should not be here

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,12 +6,54 @@ Mapbox welcomes participation and contributions from everyone.
...
+* Eliminate tiles re-creation and re-layout on zooming map with globe, when the camera is trespassing the zoom projection border. ([1891](https://github.com/mapbox/mapbox-maps-android/pull/1891))
+* Avoid tiles re-layout on enabling terrain with zero exaggeration. ([1891](https://github.com/mapbox/mapbox-maps-android/pull/1891))
+* Added API to enable/disable rendering of world copies in mercator mode. ([1891](https://github.com/mapbox/mapbox-maps-android/pull/1891))
+* Asynchronous GeoJSON data parsing when adding a new style source. ([1891](https://github.com/mapbox/mapbox-maps-android/pull/1891))
```
Comment (kiryldz) [inline CHANGELOG.md:18]: I think we should not include it here as officially we will properly land it in next beta

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,12 +6,54 @@ Mapbox welcomes participation and contributions from everyone.
...
+* Fix an issue where location updates were not provided on correct thread/looper. ([1836](https://github.com/mapbox/mapbox-maps-android/pull/1836))
+* Dispatch location updates as a location event when the app is moving to background. ([1836](https://github.com/mapbox/mapbox-maps-android/pull/1836))
+* Fix `LocationEngine` to support multiple concurrent clients. ([1836](https://github.com/mapbox/mapbox-maps-android/pull/1836))
+* Fix immediate camera animation on API level 23 or below. ([1842](https://github.com/mapbox/mapbox-maps-android/pull/1842))
 * Fix loss of `Widget` during background/foreground transition. ([1864](https://github.com/mapbox/mapbox-maps-android/pull/1864))
 * Fix displaying MapView in Android Studio Layout Preview. ([1881](https://github.com/mapbox/mapbox-maps-android/pull/1881))
 * Fix a bug where `ViewAnnotationManager.annotations` API doesn't return correct camera bounds. ([1861](https://github.com/mapbox/mapbox-maps-android/pull/1861))
```
Comment (kiryldz) [inline CHANGELOG.md:35]: btw just noticed that 😄 
```suggestion
* Fix a bug where `ViewAnnotationManager.cameraForAnnotations` API doesn't return correct camera bounds. ([1861](https://github.com/mapbox/mapbox-maps-android/pull/1861))
```

File: CHANGELOG.md (hunk 1 above)
Comment (pengdev) [reply CHANGELOG.md:36]: it's public change log entry from gl-native, I think we should include it.

Comment (kiryldz) [reply CHANGELOG.md:36]: I know but imo it was added there by mistake - it's not something new as it was backported to previous minor version

Comment (pengdev) [reply CHANGELOG.md:34]: It's new compared to v10.9.0 stable release, also patch releases are out of cycle releases and can be seriously later compared to following minor releases. I'd avoid confusions and it doesn't really matter to have one more entries, right?

Comment (kiryldz) [reply CHANGELOG.md:34]: ok, let's leave it

**Review by kiryldz — APPROVED** [review]

**Review by yunikkk — APPROVED** [review]

**Pull request merged**